generator/lib/src/dart_type_ex.dart

```diff
@@ -49,19 +50,20 @@ extension DartTypeEx on DartType {
   DartType? get nullIfDynamic => isDynamic ? null : this;
 
   DartType get basicType {
-    if (isRealmCollection) {
-      return (this as ParameterizedType).typeArguments.last;
+    final self = this;
```

Comment:
It helps with type inference later on line 54 & 55. Otherwise I would have to cast `this` on line 55 to get to `typeArguments`.

One can hope that the dart compiler becomes a bit smarter one day.